andela/ah-robotics-frontend
# #162949035 Users should be able Signup/Register to Authors Haven

#### What does this PR do?

Creates user Interfaces through which a user can be able to register and verify an account in the application.

#### Description of Task to be completed?

- [x] Create a folder structure
- [x] Create React components, views
- [x] Create React Views
- [x] Design the Register User Interface
- [x] Design Verification User Interface
- [x] Configure Redux and saga middleware
- [x] Consume the Register API
- [x] Consume Verification API

#### How should this be manually tested?

1. Clone this repository on your local machine using 

     `git clone https://github.com/andela/ah-robotics-frontend.git`
2. Change your working directory to the project's directory using 
     
     `cd ah-robotics-frontend`
3. Change to the specific branch `ft-user-register-162949035` using 

     `git checkout ft-user-register-162949035`

4. Run `npm install` to install all the modules listed in the `package.json` file.

5. Run `npm test` command to ensure the tests are passing.

    #### Testing the features
6. On the browser, head over to the URL 
`https://ah-robotics-frontend-sta-pr-12.herokuapp.com/register`, 
 The register page should be displayed as shown below.

![screen shot 2019-02-18 at 7 42 28 am](https://user-images.githubusercontent.com/10683903/52928534-471d9780-3351-11e9-9df5-47ce260e081d.png)

7. Enter the various details and click on **Register** button to create a new user. A verification email containing a link will be sent to the email provided during registration, As shown below:

<img width="832" alt="screen shot 2019-02-18 at 7 51 54 am" src="https://user-images.githubusercontent.com/10683903/52928679-23a71c80-3352-11e9-9c3d-239d58623750.png">

8. Upon clicking the link, the user will be redirected to a page displaying successful verification.

![screen shot 2019-02-18 at 7 44 53 am](https://user-images.githubusercontent.com/10683903/52928618-d32fbf00-3351-11e9-914d-3f8efb33038f.png)

#### What are the relevant pivotal tracker stories?

[#162949035](https://www.pivotaltracker.com/n/projects/2235171/stories/162949035)
